test/system/operating-systems/__snapshots__/distroless.spec.ts.snap

```diff
@@ -102,41 +102,47 @@ Object {
                 "id": "base-files@10.3+deb10u5",
                 "info": Object {
                   "name": "base-files",
+                  "purl": "pkg:deb/base-files@10.3%2Bdeb10u5",
                   "version": "10.3+deb10u5",
                 },
               },
               Object {
                 "id": "glibc/libc6@2.28-10",
                 "info": Object {
                   "name": "glibc/libc6",
+                  "purl": "pkg:deb/libc6@2.28-10?upstream=glibc",
                   "version": "2.28-10",
                 },
               },
               Object {
                 "id": "netbase@5.6",
                 "info": Object {
                   "name": "netbase",
+                  "purl": "pkg:deb/netbase@5.6",
                   "version": "5.6",
                 },
               },
               Object {
                 "id": "openssl/libssl1.1@1.1.1d-0+deb10u3",
                 "info": Object {
                   "name": "openssl/libssl1.1",
+                  "purl": "pkg:deb/libssl1.1@1.1.1d-0%2Bdeb10u3?upstream=openssl",
```

Comment:
I just added a test to our [container-sbom-export](https://github.com/snyk/container-sbom-export/blob/main/pkg/purl) library and that doesn't seem to escape the special chars:
```
pkg:deb/debian/libssl1.1@1.1.1d-0+deb10u3?upstream=openssl&distro=buster
```
I'm wondering if it should really be escaped? 🤔

Reply:
I guess it's the purl libraries that behave differently, as the TS seems to escape it and the Go one doesn't? 🥴

Reply:
The Purl spec says that "purl is just a URL", so I'd argue URL rules apply.
That means however that a plus-sign is a valid character and does *not* need to be escaped unless it's in the query component of the URL.
See [this S/O answer](https://stackoverflow.com/a/1006074) for more details.

Reply:
purl's Go library's `Parse` function calls a `url.PathUnescape` function, which mentions in it's docs that:
> PathUnescape is identical to QueryUnescape except that it does not unescape '+' to ' ' (space).

😄 

So both examples (escaped and not) work! 
I'm wondering whether the JS library behaves the same though....